SpatialGDK/Source/SpatialGDK/Public/Schema/Interest.h

```diff
@@ -9,85 +9,135 @@ namespace improbable
+{
+	EdgeLength EdgeLength;
+};
 
-	struct QueryConstraint
-	{
-		TSchemaOption<SphereConstraint> SphereConstraint;
-		TSchemaOption<CylinderConstraint> CylinderConstraint;
-		TSchemaOption<BoxConstraint> BoxConstraint;
-		TSchemaOption<RelativeSphereConstraint> RelativeSphereConstraint;
-		TSchemaOption<RelativeCylinderConstraint> RelativeCylinderConstraint;
-		TSchemaOption<RelativeBoxConstraint> RelativeBoxConstraint;
-		TSchemaOption<int64> EntityIdConstraint;
-		TSchemaOption<uint32> ComponentConstraint;
-		TArray<QueryConstraint> AndConstraint;
-		TArray<QueryConstraint> OrConstraint;
-	};
+struct QueryConstraint
+{
+	TSchemaOption<SphereConstraint> SphereConstraint;
+	TSchemaOption<CylinderConstraint> CylinderConstraint;
+	TSchemaOption<BoxConstraint> BoxConstraint;
+	TSchemaOption<RelativeSphereConstraint> RelativeSphereConstraint;
+	TSchemaOption<RelativeCylinderConstraint> RelativeCylinderConstraint;
+	TSchemaOption<RelativeBoxConstraint> RelativeBoxConstraint;
+	TSchemaOption<int64> EntityIdConstraint;
+	TSchemaOption<uint32> ComponentConstraint;
+	TArray<QueryConstraint> AndConstraint;
+	TArray<QueryConstraint> OrConstraint;
+
+	FORCEINLINE bool IsValid()
+	{
+		if (SphereConstraint.IsSet())
+		{
+			return true;
+		}
 
-	struct Query
-	{
-		QueryConstraint Constraint;
+		if (CylinderConstraint.IsSet())
+		{
+			return true;
+		}
+
+		if (BoxConstraint.IsSet())
+		{
+			return true;
+		}
+
+		if (RelativeSphereConstraint.IsSet())
+		{
+			return true;
+		}
+
+		if (RelativeCylinderConstraint.IsSet())
+		{
+			return true;
+		}
 
-		// Either full_snapshot_result or a list of result_component_id should be provided. Providing both is invalid.
-		TSchemaOption<bool> FullSnapshotResult; // Whether all components should be included or none.
-		TArray<uint32> ResultComponentId; // Which components should be included.
+		if (EntityIdConstraint.IsSet())
+		{
+			return true;
+		}
 
-		// Used for frequency-based rate limiting. Represents the maximum frequency of updates for this
-		// particular query. An empty option represents no rate-limiting (ie. updates are received
-		// as soon as possible). Frequency is measured in Hz.
-		//
-		// If set, the time between consecutive updates will be at least 1/frequency. This is determined
-		// at the time that updates are sent from the Runtime and may not necessarily correspond to the
-		// time updates are received by the worker.
-		//
-		// If after an update has been sent, multiple updates are applied to a component, they will be
-		// merged and sent as a single update after 1/frequency of the last sent update. When components
-		// with events are merged, the resultant component will contain a concatenation of all the
-		// events.
-		//
-		// If multiple queries match the same Entity-Component then the highest of all frequencies is
-		// used.
-		TSchemaOption<float> Frequency;
-	};
+		if (ComponentConstraint.IsSet())
+		{
+			return true;
+		}
+
+		if (AndConstraint.Num() > 0)
```

Comment:
Do we need to check the validity of the AndConstraints?

Reply:
Yes.